seahub/api2/endpoints/admin/devices.py

```diff
@@ -57,7 +57,11 @@ def get(self, request, format=None):
             result['platform'] = device.platform
             return_results.append(result)
 
-        return Response(({'has_next_page': has_next_page}, return_results))
+        page_info = {
+            'has_next_page': has_next_page,
+            'current_page': current_page
+        }
+        return Response((page_info, return_results))
```

Comment:
Most of the time, providing something like this in your response is sufficient for updating the pagination state.

[{"total_entries": 100}, [{}, {}, ...]]